cmd/cli/serve/serve.go

```diff
@@ -571,3 +562,51 @@ func pickP2pAddress(addresses []multiaddr.Multiaddr) multiaddr.Multiaddr {
 
 	return addresses[0]
 }
+func GetTLSCertificate(nodeConfig *node.NodeConfig, cert string, key string) (string, string, error) {
+
+	if nodeConfig.RequesterAutoCert != "" {
+		return "", "", nil
+	}
+
+	if cert != "" && key != "" {
+		return cert, key, nil
+	}
+	if cert != "" && key == "" {
+		return "", "", fmt.Errorf("invalid config: TLS cert specified without corresponding private key")
+	}
+	if cert == "" && key != "" {
+		return "", "", fmt.Errorf("invalid config: private key specified without corresponding TLS certificate")
+	}
+	if nodeConfig.RequesterSelfSigned {
+		var err error
+		// If the user has not specified a private key, use their client key
+		if key == "" {
+			key, err = config.Get[string](types.UserKeyPath)
+			if err != nil {
+				return "", "", err
+			}
+		}
+		certFile, err := os.CreateTemp(os.TempDir(), "bacalhau_cert_*.crt")
+		if err != nil {
+			return "", "", errors.Wrap(err, "unable to create temporary server certificate")
+		}
+		defer closer.CloseWithLogOnError(certFile.Name(), certFile)
+
+		var ips []net.IP = nil
+		if ip := net.ParseIP(nodeConfig.HostAddress); ip != nil {
+			ips = append(ips, ip)
+		}
+
+		if privKey, err := crypto.LoadPKCS1KeyFile(key); err != nil {
+			return "", "", err
+		} else if caCert, err := crypto.NewSelfSignedCertificate(privKey, false, ips); err != nil {
+			return "", "", errors.Wrap(err, "failed to generate server certificate")
+		} else if err = caCert.MarshalCertficate(certFile); err != nil {
+			return "", "", errors.Wrap(err, "failed to write server certificate")
+		}
+		cert = certFile.Name()
+		return cert, key, nil
+	}
```

Comment:
Instead of this huge indented block, why not return early with 

```
if !nodeConfig.RequesterSelfSigned {
    return "", "", nil 
}
```

and then carry on with the generating of the cert?